stan/math/rev/core/grad.hpp

```diff
@@ -27,8 +27,9 @@ namespace math {
  * @param vi Variable implementation for root of partial
  * derivative propagation.
  */
-static void grad(vari* vi) {
-  vi->init_dependent();
+static void grad(vari* vi = NULL) {
+  if (vi != NULL)
```

Comment:
You can just check if something is a nullptr with

```suggestion
  if (vi)
```

Since anything that's not zero will be true. Though this feels v weird, what's the intent here?

Reply:
The goal was to be able to say ```stan::math::grad(vi)``` or ```stan::math::grad()```.

I don't think it's used in this pull request though. I'm removing it.